File: backend-apps/user-service/src/server/app/route/api/bill/model/bill.model.ts

import { Schema, Model, model } from 'mongoose';
import { NextFunction } from 'express-serve-static-core';
import { BillModel } from './bill.model.interface';

export const BillSchema: Schema = new Schema({
  uuID: {
    type: String,
    required: 'uuID is required',
  },
  base64String: {
    type: String,
    required: 'base64String is required',
  },
  evaluated: {
    type: Boolean,
    default: false,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  }
});

BillSchema.pre<BillModel>("save", async function (next: NextFunction) {
  console.log("new bill created \n uuID: ", this.uuID);
  next();
});


export const Bill: Model<BillModel> = model<BillModel>("Bill", BillSchema);

File: backend-apps/ai-scanner/src/app/models/billModel.py

class BillModel:
    def __init__(self, base64String, billID, points=[
        {'y': None, 'x': None},
        {'y': None, 'x': None},
        {'y': None, 'x': None},
        {'y': None, 'x': None}
    ]):
        self.base64String = base64String
        self.billID = billID
        self.points = points

    def get(self):
        return {
            'Base64String': self.base64String,
            'billID': self.billID,
            'points': self.points,
        }
